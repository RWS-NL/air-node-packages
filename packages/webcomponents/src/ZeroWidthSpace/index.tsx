import Typography, { TypographyProps } from '@material-ui/core/Typography';
import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import React, { memo } from 'react';

export interface ZeroWidthSpaceProps extends TypographyProps {
  /** Data-qa tag to apply to the tooltip */
  'data-qa'?: string;
  /** Custom CSS classes to pass to the tooltip */
  customclasses?: string | string[];
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
}

/**
 * Inserts a Zero Width Space as a React component
 * @param props Props to pass to the Zero-Width Space
 * @example
 * ```jsx
 * <ZeroWidthSpace />
 * ```
 */
export const ZeroWidthSpace = memo(
  ({ variant, style, customclasses, 'data-qa': dataQa, ...props }: ZeroWidthSpaceProps) => (
    <Typography
      {...props}
      variant={variant || 'caption'}
      style={style}
      className={classnames(customclasses)}
      data-qa={dataQa}
    >
      &#8203;
    </Typography>
  )
);

export default ZeroWidthSpace;
