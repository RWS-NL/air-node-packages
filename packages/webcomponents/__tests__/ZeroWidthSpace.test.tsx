import { shallow } from 'enzyme';
import ZeroWidthSpace from '../src/ZeroWidthSpace/ZeroWidthSpace';
import React from 'react';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const zeroWidthSpace = shallow(<ZeroWidthSpace />);
    expect(zeroWidthSpace).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const zeroWidthSpace = shallow(<ZeroWidthSpace />);
    expect(zeroWidthSpace).toMatchSnapshot();
  });

  test('AdditionalTest', () => {
    const zeroWidthSpace = shallow(<ZeroWidthSpace />);
    expect(zeroWidthSpace).toMatchSnapshot();
  });
});