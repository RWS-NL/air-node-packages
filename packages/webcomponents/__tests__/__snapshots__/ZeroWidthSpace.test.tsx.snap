// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing AdditionalTest 1`] = `
<WithStyles(ForwardRef(Typography))
  className=""
  variant="caption"
>
  ​
</WithStyles(ForwardRef(Typography))>
`;

exports[`Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(Typography))
  className=""
  variant="caption"
>
  ​
</WithStyles(ForwardRef(Typography))>
`;

exports[`Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(Typography))
  className=""
  variant="caption"
>
  ​
</WithStyles(ForwardRef(Typography))>
`;
